diffrax/brownian/path.py

```diff
@@ -43,20 +68,29 @@ def t1(self):
         return None
 
     @eqx.filter_jit
-    def evaluate(self, t0: Scalar, t1: Scalar, left: bool = True) -> Array:
+    def evaluate(self, t0: Scalar, t1: Scalar, left: bool = True) -> PyTree[Array]:
         del left
         nondifferentiable_input(t0, "t0")
         nondifferentiable_input(t1, "t1")
         t0_ = force_bitcast_convert_type(t0, jnp.int32)
         t1_ = force_bitcast_convert_type(t1, jnp.int32)
         key = jrandom.fold_in(self.key, t0_)
         key = jrandom.fold_in(key, t1_)
-        return jrandom.normal(key, self.shape) * jnp.sqrt(t1 - t0)
+        key = split_by_tree(key, self.shape)
+        return jtu.tree_map(
+            lambda key, shape: self._evaluate_leaf(t0, t1, key, shape), key, self.shape
+        )
+
+    def _evaluate_leaf(self, t0: Scalar, t1: Scalar, key, shape: jax.ShapeDtypeStruct):
+        return jrandom.normal(key, shape.shape, shape.dtype) * jnp.sqrt(t1 - t0).astype(
+            shape.dtype
+        )
 
 
 UnsafeBrownianPath.__init__.__doc__ = """
 **Arguments:**
 
-- `shape`: What shape each individual Brownian sample should be.
+- `shape`: What shape each individual Brownian sample should be. Can be 
+specified by a pytree.
```

Comment:
Can this be a little more explicit?

```
Should be a PyTree of `jax.ShapeDtypeStruct`s, representing the shape, dtype, and PyTree structure of the output. For simplicity, `shape` can also just be a tuple of integers, describing the shape of a single JAX array.
```

Reply:
You're right. Also added

```
... For simplicity, `shape` can also just be a tuple of integers, describing the shape of a single JAX array. In that case the dtype is chosen to be `float64` if `JAX_ENABLE_X64=True` and `float32` otherwise.
```

which I _think_ is correct